javascript/hasOwnProperty.md

```diff
@@ -0,0 +1,48 @@
+hasOwnProperty() is a default method in the Object prototype that takes a string argument and returns a boolean indicating whether the object has the specified property.
+
+	obj.hasOwnProperty(prop);
+
+The hasOwnProperty determines whether the properties defined in object are a direct property of that object as opposed to checking its prototypal inheritance.
+
+Similarly, the 'in' operator checks to see if a particular attribute is located anywhere in the object, however, this is inclusive of its prototype chain.
+
+##Direct versus Indirect
+
+	var object1 = {prop1: {c:1, d:2}, prop2:[1,2,3]};
```

Comment:
Instead of formatting the code with 4 spaces (which is ok), you could start with 3 back-tildes like this: \`\`\`. This way you can use syntax highlighting for a specific language, in this case JS. You can do this:
\`\`\`JavaScript
(your code here)
\`\`\`
It is important that the opening and closing back-tildes are on their own line, the opening one with the language you want to highlight, as I wrote in the example